index.rst

```diff
@@ -6,24 +6,474 @@
+
+::
+
+  warnings.simplefilter("ignore", category=UserWarning)
+
+
+Markdown style
+--------------
+
+Any references to variables used in code cells or any code commands should be in ``monospaced font``.
+
+Use of indented text should be limited to warnings and notices, e.g., ``> **Warning:** the following cell...``. 
+
+*The most appropriate narrative voice for tutorials is still under consideration.*
+Adopt your own style for now.
+
+
+Code cell comments
+------------------
+
+Avoid using comments within a code cell as documentation.
+Markdown cells are the preferred way to provide descriptive text.
+
+
+Code cell style standard PEP8
+-----------------------------
+
+``PEP8`` is the standard, and ``flake8`` is a tool to ensure compliance with the standards.
+
+Use ``flake8`` to ensure notebook code conforms to codebase style `PEP8 <https://www.python.org/dev/peps/pep-0008/>`_ , with a few exceptions. 
+
+Notebook tutorial developers must install the following packages locally in their home directory:
+
+::
+
+  pip install --user flake8-nb
+  pip install --user pycodestyle_magic
+
+It is known that the most up-to-date version of flake8 has some issues.
+If errors are encountered such as "AttributeError: '_io.StringIO' object has no attribute 'buffer'", 
+force-downgrade ``flake8`` from vesion 4.0.1 to 3.9.2 with ``pip install flake8=3.9.2``.
+
+
+The flake8 config file
+^^^^^^^^^^^^^^^^^^^^^^
+
+Create a configuration file for ``flake8``.
+
+These instructions use emacs but it doesn’t matter so long as the end result is correctly-named file with the right contents. 
+For example, from the command line in your home directory, execute:
+
+::
+
+  touch .config/flake8
+  emacs .config/flake8
+
+
+Then copy-paste the following into the opened config file:
+
+::
+
+  [flake8]
+  max-line-length = 99
+  ignore = E133, E226, E228, E266, N802, N803, N806, N812, N813, N815, N816, W503
+
+Use ``x-s`` then ``x-c`` to save and exit emacs.
+
+
+While developing a notebook
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While developing a notebook, have the following 'magic' commands as the first code cell:
+
+::
+
+  %load_ext pycodestyle_magic
+  %flake8_on
+  import logging
+  logging.getLogger("flake8").setLevel(logging.FATAL)
+
+Whenever you execute a cell, it will use ``flake8`` to check for adherence to the ``PEP8`` coding style guide, and report violations.
+Fix them as you go.
+Once you're done with the entire notebook you can remove that cell with the magic commands. 
+
+
+When the notebook is complete
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When the notebook is complete, from the command line in the notebook's directory execute:
+
+::
+
+  flake8-nb notebook_name.ipynb
+
+
+This will give you a final check of any violations with ``PEP8``.
+This will catch things that can be missed line-by-line, such as packages that are imported but never used.  
+
+
+Git branch, merge, and review policy for ``tutorial-notebooks``
```

Comment:
If possible, in the instructions below, we should write down the explicit git commands we'd like people to use (as in the DM Developers Guide).

Reply:
At this time, we want to be flexible to the tools people are using (e.g. web UI v. command line)